src/wp-includes/canonical.php

```diff
@@ -547,6 +547,65 @@ function redirect_canonical( $requested_url = null, $do_redirect = true ) {
 		}
 	}
 
+	/**
+	 * Pretty permalinks for search-results were removed in 6.6.0.
+	 * All search results are now served from /?s={search_query}.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/4463
+	 */
+	if ( str_contains( $requested_url, '/search/' ) && ( is_404() || is_feed() ) ) {
```

Comment:
This is a little imprecise for comment feeds of content with the slug `search`

1. Set permalinks to year/month/post-name
2. Create a post with the slug `search`
3. Go to the page `yyyy/mm/search/feed`
4. At the end of this condition the redirect URL is set to `/?s=feed`

At present it doesn't do a redirect but it would be nice not to enter this block of code in that situation. This will help avoid the potential for bugs later.